VMEncryption/mdc-recommendation/helper-scripts/VerifyVMSupport.py

```diff
@@ -1,164 +1,164 @@
+    is_databricks_vm = False 
+
+    if (microsoft_aks_image_publisher in vm_image_publisher) or (vm_image_sku.startswith(aks_prefix)):
+        is_databricks_vm = True
+        reasons_phrase_str += "Databricks VM is not supported for ADE or HBE encryption."
+    elif (microsoft_databrick_image_publisher in vm_image_publisher):
+        is_databricks_vm = True
+
+    return is_databricks_vm
+
+def is_vm_using_ultra_disks(imds_metadata):
+    vm_data_disks = imds_metadata['compute']['storageProfile']['dataDisks']
+    is_vm_using_ultra_disk = False
+
+    if len(vm_data_disks) > 0:    
+        for disk in vm_data_disks:
+            ultra_disk_flag = disk['isUltraDisk']
+            if (ultra_disk_flag is not None) and ("true" in str(ultra_disk_flag).lower()):
+                is_vm_using_ultra_disk = True
+                break
+  
+    return is_vm_using_ultra_disk
+
+# The below method covers a small intersection of scenarios where the OS distro is unsupported by ADE and VM size is unsupported by HBE.
+def is_os_vmsize_unsupported(imds_metadata):
+    unsupported_os_and_vm_size = False 
+    vm_os_offer = imds_metadata['compute']['storageProfile']['imageReference']['offer']
+    vm_os_sku = imds_metadata['compute']['storageProfile']['imageReference']['sku']
+    vm_image_tuple = tuple((vm_os_offer, vm_os_sku))
+    vm_size = imds_metadata['compute']['vmSize']
+
+    if vm_size is not None:
+        if (vm_image_tuple not in supported_ade_image_list) and (vm_size in unsupported_vm_sku_hbe_encryption):
+            unsupported_os_and_vm_size = True
+        
+    return unsupported_os_and_vm_size
+    
+def check_vm_supportability(imds_metadata):
+    #check if VM is supported for ADE or HBE encryption
+    is_vm_not_supported = False
+    reasons_code = ""
+    reasons_phrase_str = ""
+    vm_support_dict = {}
+    
+    if is_basic_vm(imds_metadata):
+        is_vm_not_supported = True
+        reasons_code = "BasicSizedVirtualMachine"
+        reasons_phrase_str += "VM size is not supported by Azure Disk Encryption and EncryptionAtHost."
+
+    if is_classic_vm(imds_metadata):
+        is_vm_not_supported = True
+        reasons_code = "ClassicVirtualMachine"
+        reasons_phrase_str += "Classic VM is not supported by Azure Disk Encryption and EncryptionAtHost."
+
+    if is_vm_using_ultra_disks(imds_metadata):
+        is_vm_not_supported = True
+        reasons_code = "UltraDisksVirtualMachine"
+        reasons_phrase_str += "VM is using ultra disks which is currently not supported by Azure Disk Encryption and EncryptionAtHost."
+
+    if verify_vm_os_sku_completeness(imds_metadata) is False:
+        vm_support_dict['is_vm_not_supported'] = True
+        vm_support_dict['code'] = "UnknownOsVirtualMachine"
+        vm_support_dict['phrase'] = "VM OS details are missing in IMDS metadata for verification"
+        return vm_support_dict
+
+    else:
```

Comment:
Is this else condition required? The function will anyway return if the above if condition is True.

Reply:
Agree, removed the else condition in latest commit.